kubernetes/eventhandler.go

```diff
@@ -134,19 +138,27 @@ type podUpdaterStore interface {
 	List() []interface{}
 }
 
+// OldobjectWatcher is the interface  that an object needs to implement to be
+// able to use Oldobject cache event function from watcher.
+type OldobjectWatcher interface {
+	Oldobject() runtime.Object
+}
+
 // namespacePodUpdater notifies updates on pods when their namespaces are updated.
 type namespacePodUpdater struct {
-	handler podUpdaterHandlerFunc
-	store   podUpdaterStore
-	locker  sync.Locker
+	handler          podUpdaterHandlerFunc
+	store            podUpdaterStore
+	namespacewatcher OldobjectWatcher
+	locker           sync.Locker
 }
 
 // NewNamespacePodUpdater creates a namespacePodUpdater
-func NewNamespacePodUpdater(handler podUpdaterHandlerFunc, store podUpdaterStore, locker sync.Locker) *namespacePodUpdater {
+func NewNamespacePodUpdater(handler podUpdaterHandlerFunc, store podUpdaterStore, namespacewatcher OldobjectWatcher, locker sync.Locker) *namespacePodUpdater {
 	return &namespacePodUpdater{
-		handler: handler,
-		store:   store,
-		locker:  locker,
+		handler:          handler,
+		store:            store,
+		namespacewatcher: namespacewatcher,
```

Comment:
can we use MixedCaps (aka camelCase) here and for all other variables? as suggested at https://go.dev/doc/effective_go?

Reply:
Update in https://github.com/elastic/elastic-agent-autodiscover/pull/71/commits/16b0fca86b19f8f8f108c219f43f1024687da2ca

Let me know If I missed anything